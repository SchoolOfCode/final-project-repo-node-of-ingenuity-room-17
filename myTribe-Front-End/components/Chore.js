import {
  StyleSheet,
  Text,
  View,
  ImageBackground,
  Image,
  TextInput,
  Button,
  KeyboardAvoidingView,
  Dimensions,
  ScrollView,
  TouchableOpacity,
} from "react-native";

const Chore = ({
  title,
  member,
  dueDate,
  description,
  isComplete,
  completeChore,
  id,
}) => {
  return (
    <TouchableOpacity
      onLongPress={() => {
        completeChore(id);
      }}
      style={isComplete ? styles.viewCompleted : styles.chore}
    >
      <Text style={isComplete ? styles.titleCompleted : styles.title}>
        {title}
      </Text>
      <Text style={isComplete ? styles.memberCompleted : styles.member}>
        {member}
      </Text>
      <Text style={isComplete ? styles.dateCompleted : styles.dueDate}>
        {dueDate}
      </Text>
      <Text
        style={isComplete ? styles.descriptionCompleted : styles.description}
      >
        {description}
      </Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  chore: {
    width: "100%",
    marginBottom: 10,
    position: "relative",
    padding: 10,
    backgroundColor: "white",
    shadowColor: "black",
    shadowOffset: { width: 0, height: 4 },
    shadowRadius: 2,
    shadowOpacity: 0.1,
  },

  dueDate: {
    position: "absolute",
    top: 10,
    right: 10,
    color: "#FFBD00",
  },

  title: {
    fontSize: 18,
    fontWeight: "bold",
  },

  member: {
    color: "#FFBD00",
    marginBottom: 20,
    fontWeight: "bold",
    marginTop: 5,
  },

  description: {
    color: "#AAAAAC",
  },
  viewCompleted: {
    backgroundColor: "#FFBD00",
    color: "white",
  },
  titleCompleted: {
    backgroundColor: "#FFBD00",
    color: "white",
  },
  memberCompleted: {
    backgroundColor: "#FFBD00",
    color: "white",
  },
  descriptionCompleted: {
    backgroundColor: "#FFBD00",
    color: "white",
  },
  dateCompleted: {
    backgroundColor: "#FFBD00",
    color: "white",
  },
});

export default Chore;
