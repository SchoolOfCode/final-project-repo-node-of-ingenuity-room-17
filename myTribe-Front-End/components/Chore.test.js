import React from "react";
import { View } from "react-native";
import renderer from "react-test-renderer";

import Chore from "./Chore";

//Test to check if the chore component is being rendered
describe("Check if chore component has 4 children ", () => {
  it("has 4 children", () => {
    const tree = renderer.create(<Chore />).toJSON();
    expect(tree.children.length).toBe(4);
  });

  //Test to check if what is being rendered in chore component matches the snapshop in "_snapshot" file
  it("Check if chore component renders correctly", () => {
    const tree = renderer.create(<Chore />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
