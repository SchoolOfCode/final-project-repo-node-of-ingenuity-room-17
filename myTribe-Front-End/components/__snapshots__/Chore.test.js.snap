// Jest Snapshot v1, https://goo.gl/fbAQLP


exports[`Check if chore component has 4 children  Check if chore component renders correctly 1`] = `

<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",


      "marginBottom": 10,
      "opacity": 1,
      "padding": 10,
      "position": "relative",
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 2,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#FFBD00",
        "fontWeight": "bold",
        "marginBottom": 20,
        "marginTop": 5,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#FFBD00",
        "position": "absolute",
        "right": 10,
        "top": 10,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#AAAAAC",
      }
    }
  />
</View>
`;
